### PR TITLE
#61 Review contracts marked for removal.

### LINKED ISSUE #61: SelfPmApplication.java: Write a scheduled method whic...
The puzzle ``55-4af5fb61`` originating from #55 has to be resolved:

https://github.com/self-xdsd/self-pm/blob/97637be5e8ce832b66d7d1db4f60c45eb455b275/src/main/java/com/selfxdsd/selfpm/SelfPmApplication.java#L34-L38

"Write a scheduled method which will make each PM review and completely remove any Contract which has been marked for removal more than 30 days ago (if markedForRemoval > 30 days). This method should run once per day and start with a delay of 15 minutes, so it will not overlap with other scheduled jobs.".

It was created by @amihaiemil on 2020-11-09T09:05:58Z. Estimation is ``60 minutes``.
If you have any technical questions, don't ask me, I won't be able to help. Open new issues instead.

### PR DESCRIPTION
**Pull request by criske**
Closes #61 

- left puzzle to proper remove a Contract when API is available.

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-11-12T19:47:22.155735``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/java/com/selfxdsd/selfpm/ReviewContractsMarkedForRemoval.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                    + "... "
+                );
+                final List<Contract> toRemove = this.contractsToRemove(project);
+                LOG.debug(
+                    "For project " + project.repoFullName() + " at "
+                        + project.provider() + " there are " + toRemove.size()
+                        + " contracts that will be removed..."
+                );
+                for(final Contract contract: toRemove){
+                    project.resolve(this.contractRemovalEvent(contract));
```
Comment (criske) [inline src/main/java/com/selfxdsd/selfpm/ReviewContractsMarkedForRemoval.java:131]: @amihaiemil I've used here the Event API to verify in tests which contracts are removed,
until we have an API for that.

Comment (criske) [reply src/main/java/com/selfxdsd/selfpm/ReviewContractsMarkedForRemoval.java:131]: @amihaiemil On a 2nd thought, using the events api in a way that wasn't intented is a bad idea, luckly `StoredProjec`t dosen't throw if the event is unknown. Just injecting a simple callback in testing constructor is much more clearer aproach.

**Comment by amihaiemil** [comment]
@criske looks elegant, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**